include/onnxruntime/core/graph/graph.h

```diff
@@ -1025,6 +1109,12 @@ class Graph {
   // All node args owned by <*this> graph. Key is node arg name.
   std::unordered_map<std::string, std::unique_ptr<NodeArg>> node_args_;
 
+  // node arg to its producer node
+  std::unordered_map<std::string, NodeIndex> node_arg_to_producer_node_;
```

Comment:
I suggest not to store this information since it can be deduced from graph. For example, in transformers, user could delete nodes, add nodes, add node_arg, reuse node_arg in a new node. All these changes will impact the consistent of this map with the graph.

If needed, we can expose a function to retrieve such a map.

Reply:
will address this later, created a work item to track it

---
In reply to: [414817323](https://github.com/microsoft/onnxruntime/pull/3681#discussion_r414817323) [](ancestors = 414817323)